### DISCUSSION TITLE
One Tiff image: two page PDF after conversion

### DISCUSSION THREAD

**Discussion by Cicorione** [ImageMagick/ImageMagick6#282] [General] [answered]
### ImageMagick version

6.9.11-60 Q16

### Operating system, version and so on

Devuan Daedalus AMD64

### Description

Hello,

I am trying to convert some Tiff files in PDF to pass to a LuaTeX based DTP, but the process fails because the conversion generates two pages instead of one.

Here the most basic example I can reproduce:
```
convert -verbose tiff.tif  tiff-no-absolute-path.pdf
tiff.tif[0] TIFF 241x117 241x117+28+89 8-bit Grayscale Gray 9640B 0.010u 0:00.004
tiff.tif[1] TIFF 295x295 295x295+0+0 8-bit TrueColor sRGB 0.010u 0:00.003
```

These are the files:
[tiff.zip](https://github.com/ImageMagick/ImageMagick/files/12987565/tiff.zip)

Thanks! 🙏

**Comment by fmw42** [comment, marked as answer]
The tiff file has two images. Use the first by adding [0] to the end of the file name. Also there is a large internal offset, so you need to add +repage to the command

magick tiff.tif[0] +repage x.pdf

[x.pdf](https://github.com/ImageMagick/ImageMagick6/files/13030649/x.pdf)

**Reply by Cicorione** [reply]
@fmw42 thank you very much.

Just to confirm:

Is the first image the one used as thumbnail?
And is the second image the actual image?

**Reply by fmw42** [reply]
I do not know what you mean by thumbnail.  There are two images in the file.  The first image is slightly smaller than the second. But not a large difference.  The first image is the one I used and is selected via the [0] at the end of the file name. The second image is actually in color for the text. (It would be selected using [1] at the end of the file name).

**Reply by Cicorione** [reply]
Oh... I see, you are just smart.

So basically, Tiff can have multiple instance of the same image:

> A baseline TIFF file can contain a sequence of images (IFD). Typically, all the images are related but represent different data, such as the pages of a document. In order to explicitly support multiple views of the same data, the SubIFD tag was introduced.[[9]](https://en.wikipedia.org/wiki/TIFF#cite_note-supplement1-9) This allows the images to be defined along a [tree structure](https://en.wikipedia.org/wiki/Tree_structure). Each image can have a sequence of children, each child being itself an image. The typical usage is to provide thumbnails or several versions of an image in different color spaces.

(https://en.wikipedia.org/wiki/TIFF)

So basically Tiff has stored the preview of the image that you can see from you file browser, I knew it but I didn't know it could affect the conversion. This was my first attempt to convert Tiff to PDF through Image Magick, and it did the conversion properly, that's why I wrote that you're actually very smart!

The Tiff format is quite arcane and unused even though still valid to store hires images, but having so much trouble to convert in PDF is undesirable, but your contribute for me was HUGE!

Thank very much again! 🙏

**Reply by fmw42** [reply]
I do not know how your TIFF file was created. It could actually have two different images.  The first frame Is not a preview in the sense that it is a smaller version of the larger file. But it is the one you see mostly when looking at the file. The actual preview is likely created from the first frame to show in the file icon.
